File: 2414-move-pieces-to-obtain-a-string/move-pieces-to-obtain-a-string.cpp
class Solution {
public:
    bool canChange(string start, string target) {
        int i = 0, j = 0;
        while (i < start.length() || j < target.length()){
            cout << i << " " << j <<"\n";
            while (start[i] == '_')
                i++;
            while (target[j] == '_')
                j++;
            if (i == start.length() || j == target.length())
                return i == start.length() && j == target.length();
            if (start[i] != target[j] || start[i] == 'L' && i < j || start[i] == 'R' && i > j)
                return false;
            i++;
            j++;             
        }
        return true;
    }
};

File: 2414-move-pieces-to-obtain-a-string/move-pieces-to-obtain-a-string.py
class Solution:
    def canChange(self, start: str, target: str) -> bool:
        i, j = 0, 0
        while i < len(start) and j < len(target):
            if start[i] == target[j] == '_':
                i += 1
                j += 1
            elif start[i] == '_' :
                i += 1
            elif target[j] == '_':
                j += 1
            elif start[i] == target[j] == 'R':
                if i > j:
                    return False
                else:
                    i += 1
                    j += 1
            elif start[i] == target[j] == 'L':
                if i < j:
                    return False
                else:
                    i += 1
                    j += 1
            else:
                return False
        
        while i < len(start) and start[i] == '_' :
                i += 1

        while j < len(target) and target[j] == '_':
                j += 1
        return True if i == len(start) and j == len(target) else False